### PR TITLE
BUG/TST:  fillna limit checks and tests

### LINKED ISSUE #27042: fillna not throwing error when limit is not positive
#### Code Sample

```python
import pandas as pd
import numpy as np
# Your code here
NCOLS = 3
NROWS = 50
data = {"col{}".format(int((i - NCOLS / 2) % NCOLS + 1)): np.random.randint(
            -100, 100, size=(NROWS)
        )
        for i in range(NCOLS)
}
df = pd.dataframe(data)
df.fillna(0, limit=-5)
```
#### Problem description

I am currently on pandas 0.24.2 and running the above code does not throw an error even though the limit is less than or equal to zero. This error does not arise when the `data` is made up of floats instead of ints (replace `np.random.randint` with `np.random.uniform`).

#### Expected Output

`ValueError: Limit must be greater than 0`

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.6.1.final.0
python-bits: 64
OS: Darwin
OS-release: 18.5.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.24.2
pytest: 4.3.0
pip: 19.0.3
setuptools: 40.8.0
Cython: 0.29.7
numpy: 1.15.0
scipy: 1.2.1
pyarrow: 0.14.0.RAY
xarray: 0.12.0
IPython: 7.3.0
sphinx: None
patsy: None
dateutil: 2.8.0
pytz: 2018.9
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: 3.0.2
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml.etree: None
bs4: None
html5lib: None
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.10
s3fs: 0.2.0
fastparquet: None
pandas_gbq: None
pandas_datareader: None
gcsfs: None
</details>

### PR DESCRIPTION
**Pull request by Inevitable-Marzipan**
Reversed the order of returning and performing checks on `limit`
in fillna method in blocks.py so that an ValueError is raised
before returning.

Alternative to PR #27074

- [tick ] closes #27042 
- [ ] tests added / passed
- [tick ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ tick] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v0.25.0.rst [hunk 1]
```diff
@@ -685,7 +685,8 @@ Missing
 
 - Fixed misleading exception message in :meth:`Series.interpolate` if argument ``order`` is required, but omitted (:issue:`10633`, :issue:`24014`).
 - Fixed class type displayed in exception message in :meth:`DataFrame.dropna` if invalid ``axis`` parameter passed (:issue:`25555`)
--
+- Fixed not throwing ValueError in :meth:`DataFrame.fillna` when called with a negative ``limit`` parameter value (:issue:`27042`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.25.0.rst]: can you put ValueError in double back quotes

**Review by jreback — CHANGES_REQUESTED** [review]
i like this a lot better thatn the other one :->

File: doc/source/whatsnew/v0.25.0.rst (hunk 1 above)
Comment (Inevitable-Marzipan) [reply doc/source/whatsnew/v0.25.0.rst]: done

File: doc/source/whatsnew/v0.25.0.rst [hunk 2]
```diff
@@ -685,6 +685,7 @@ Missing
 
 - Fixed misleading exception message in :meth:`Series.interpolate` if argument ``order`` is required, but omitted (:issue:`10633`, :issue:`24014`).
 - Fixed class type displayed in exception message in :meth:`DataFrame.dropna` if invalid ``axis`` parameter passed (:issue:`25555`)
+- Fixed not throwing ``ValueError`` in :meth:`DataFrame.fillna` when called with a negative ``limit`` parameter value (:issue:`27042`)
```
Comment (WillAyd) [inline doc/source/whatsnew/v0.25.0.rst]:
```suggestion
- A ``ValueError`` will now be thrown by :meth:`DataFrame.fillna` when ``limit`` is not a positive integer (:issue:`27042`)
```

Minor suggestion on wording to simplify and make more accurate

**Review by WillAyd — CHANGES_REQUESTED** [review]
Great change!

**Review by jreback — APPROVED** [review]

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Thanks @Inevitable-Marzipan !

**Comment by Inevitable-Marzipan** [comment]
Thanks for the help guys, 1st PR done!

**Pull request merged**